cli.py

```diff
@@ -0,0 +1,166 @@
+import sys
+import argparse
+import bech32m
+import enc_dec
+
+
+def user_interface():
+    """
+    provides cli and stdin/stdout options to user for encoding/decoding
+    :return: mode, hrp, input_string, infile, input_format, output_format, outfile
+    """
+
+    if len(sys.argv) > 1:
+        my_parser = argparse.ArgumentParser(description='Bech32m Encoder/Decoder')
+        my_parser.add_argument('m', type=str, help='encode/decode', choices=['enc', 'dec'])
+        my_parser.add_argument('hrp', type=str, help='human readable part[ASCII]')
+        group = my_parser.add_mutually_exclusive_group(required=True)
+        group.add_argument('-s', type=str, help='input string to be encoded/decoded')
+        group.add_argument('-inf', type=str, help='input file to read string to be encoded/decoded')
+        my_parser.add_argument('-iform', type=str, choices=['hex', 'bin', 'b64'], help='input format')
+        my_parser.add_argument('-oform', type=str, choices=['hex', 'bin', 'b64'], help='output format')
+        my_parser.add_argument('-of', type=str, help='output file path to write encoded/decoded string')
+        args = my_parser.parse_args()
+
+        if args.s is not None and args.iform == "bin":
+            args_s = args.s
+            args_str = args_s.encode()
+        else:
+            args_str = args.s
+
+        return args.m, args.hrp, args_str, args.inf, args.iform, args.oform, args.of
```

Comment:
suggestion (non-blocking): This is a lot of related values, maybe return them in a single object.